### PR TITLE
add useHeader, delimiter, quote args to csvFile; Also CsvRelation needs ...

### PR DESCRIPTION
**Pull request by huangjs**
...to extend BaseRelation

### CODE REVIEW

**Comment by huangjs** [comment]
Looks like sbt-launcher jar download failed in Travis CI.

$ sbt -jvm-opts travis/jvmopts.test scalastyle
Detected sbt version 0.13.6
Downloading sbt launcher for 0.13.6:
  From  http://typesafe.artifactoryonline.com/typesafe/ivy-releases/org.scala-sbt/sbt-launch/0.13.6/sbt-launch.jar
    To  /home/travis/.sbt/launchers/0.13.6/sbt-launch.jar
Download failed. Obtain the jar manually and place it at /home/travis/.sbt/launchers/0.13.6/sbt-launch.jar


On my laptop, `sbt scalastyle` succeeded.

?feature-csvfile-with-options ~/workspace/spark/spark-csv> sbt  scalastyle 
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8
Listening for transport dt_socket at address: 5008
[info] Loading global plugins from /Users/jianshuang/.sbt/0.13/plugins
[info] Loading project definition from /Users/jianshuang/workspace/spark/spark-csv/project
[info] Set current project to spark-csv (in build file:/Users/jianshuang/workspace/spark/spark-csv/)
[info] scalastyle using config /Users/jianshuang/workspace/spark/spark-csv/scalastyle-config.xml
[info] Processed 4 file(s)
[info] Found 0 errors
[info] Found 0 warnings
[info] Found 0 infos
[info] Finished in 6 ms
[success] created output: /Users/jianshuang/workspace/spark/spark-csv/target
[success] Total time: 1 s, completed Mar 10, 2015 4:52:34 PM


Jianshi

File: src/main/scala/com/databricks/spark/csv/package.scala [hunk 1]
```diff
@@ -23,19 +23,22 @@ package object csv {
    * Adds a method, `csvFile`, to SQLContext that allows reading CSV data.
    */
   implicit class CsvContext(sqlContext: SQLContext) {
-    def csvFile(filePath: String) = {
+    def csvFile(filePath: String,
```
Comment (falaki) [inline src/main/scala/com/databricks/spark/csv/package.scala]: This implicit function is for the common case of of comma separated files (similar to tab separated files). That is why delimiter is hardcoded to ',' 
Similarly for the very common case we hard coded quote to '"'

Comment (huangjs) [inline src/main/scala/com/databricks/spark/csv/package.scala]: Common case is handled with default parameters. Why should we limit the options to users?

**Comment by falaki** [comment]
Would you please rebase this?

**Comment by huangjs** [comment]
done.

On Wed, Mar 18, 2015 at 1:07 AM, Hossein Falaki <notifications@github.com>
wrote:

> Would you please rebase this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/databricks/spark-csv/pull/26#issuecomment-82483986>.
>



-- 
Jianshi Huang

LinkedIn: jianshi
Twitter: @jshuang
Github & Blog: http://huangjs.github.com/

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala [hunk 2]
```diff
@@ -61,6 +61,32 @@ class CsvSuite extends FunSuite {
     assert(results.size === 2)
   }
 
+  test("DSL test with alternative delimiter and quote using sparkContext.csvFile") {
+    val results =
+      TestSQLContext.csvFile(carsAltFile, useHeader = true, delimiter = '|', quote = ''')
```
Comment (falaki) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: This should be '\'' (middle single quote needs to be escaped.)

**Comment by falaki** [comment]
@huangjs I left a comment on the test, and then this is good to go.

**Comment by huangjs** [comment]
Fixed.

On Thu, Mar 26, 2015 at 2:55 AM, Hossein Falaki <notifications@github.com>
wrote:

> @huangjs <https://github.com/huangjs> I left a comment on the test, and
> then this is good to go.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/databricks/spark-csv/pull/26#issuecomment-86171172>.
>



-- 
Jianshi Huang

LinkedIn: jianshi
Twitter: @jshuang
Github & Blog: http://huangjs.github.com/

**Comment by falaki** [comment]
Thanks! Merging this.

**Pull request merged**